src/pb_stub.cc

```diff
@@ -845,11 +846,19 @@ PYBIND11_EMBEDDED_MODULE(c_python_backend_utils, module)
   py::class_<InferRequest, std::shared_ptr<InferRequest>>(
       module, "InferenceRequest")
       .def(
-          py::init<
-              const std::string&, uint64_t,
-              const std::vector<std::shared_ptr<PbTensor>>&,
-              const std::vector<std::string>&, const std::string&,
-              const int64_t, const uint32_t>(),
+          py::init([](const std::string& request_id, uint64_t correlation_id,
+                      const std::vector<std::shared_ptr<PbTensor>>& inputs,
+                      const std::vector<std::string>& requested_output_names,
+                      const std::string& model_name,
+                      const int64_t model_version, const uint32_t flags) {
+            std::set<std::string> requested_outputs;
+            for (auto& requested_output_name : requested_output_names) {
+              requested_outputs.emplace(requested_output_name);
+            }
+            return std::make_shared<InferRequest>(
+                request_id, correlation_id, inputs, requested_outputs,
+                model_name, model_version, flags);
+          }),
```

Comment:
nit: `requested_output_names`

Reply:
Also, why not use `std::set`? Is it because pybind doesn't support?

Reply:
@jbkyang-nvi requested_output_names will create a conflict because a variable with the same name already exists.

>why not use std::set? Is it because pybind doesn't support?

The reason is that it will be a breaking change to the user's Python model. Currently they are expected to pass a list. Changing it to a std::set will required them providing a Python set.

Reply:
I would recommend changing the name to `requested_output_names_non_unique` instead. Do we have a plan of when we want to have the users change their model? If so, we should add a ticket to remind us of this workaround

Reply:
I think the name is too long. If you have shorter variable name suggestions I'd be happy to change it. We neither have any plans or want users to change their Python model. This is only an internal change.